### PR TITLE
Convert most of the rest of the plan-generator-tests into cgtest format.

### PR DESCRIPTION
**Pull request by shans**
There's a couple I haven't done, either because the difference seemed pretty marginal or because there were extra complications (e.g. multiple input files instead of one).

Also note the new resolved singleton typeVariable test. It's currently showing a bug in that the Entity type isn't wrapped in a Singleton. Fix coming v. soon!

### CODE REVIEW

File: src/tools/tests/goldens/plan-generator-handleConnections.cgtest [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                queryExpression = true.asExpr()
+            )
+        )
+    ),
+    emptyList()
+)
+-----[end]-----
+
+-----[name]-----
+generates schemas for resolved type variable singletons
```
Comment (shans) [inline src/tools/tests/goldens/plan-generator-handleConnections.cgtest:85]: new test here.

File: src/tools/tests/goldens/plan-generator-handleConnections.cgtest [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    R_Handle0,
+    HandleMode.Write,
+    arcs.core.data.SingletonType(arcs.core.data.EntityType(B_Data.SCHEMA)),
+    emptyList()
+)
+-----[next]-----
+HandleConnection(
+    R_Handle0,
+    HandleMode.Read,
+    arcs.core.data.EntityType(
```
Comment (shans) [inline src/tools/tests/goldens/plan-generator-handleConnections.cgtest:120]: this should be wrapped in a SingletonType

File: src/tools/tests/codegen-unit-test-base.ts [hunk 3]
```diff
@@ -94,7 +95,8 @@ export abstract class ManifestCodegenUnitTest extends CodegenUnitTest {
  */
 export async function runCompute(testCase: CodegenUnitTest, test: Test): Promise<string[]> {
   Flags.reset();
-  const result = await testCase.compute(test.input, test.options, test);
+DriverFactory.clearRegistrationsForTesting();
```
Comment (piotrswigon) [inline src/tools/tests/codegen-unit-test-base.ts:98]: 2 spaces of indent

File: src/tools/tests/goldens/plan-generator-handleConnections.cgtest (hunk 2 above)
Comment (piotrswigon) [reply src/tools/tests/goldens/plan-generator-handleConnections.cgtest:120]: #5846

File: src/tools/tests/goldens/plan-generator-handleConnections.cgtest (hunk 1 above)
Comment (piotrswigon) [reply src/tools/tests/goldens/plan-generator-handleConnections.cgtest:85]: Thank you 🙏

File: src/tools/tests/goldens/plan-generator-handles.cgtest [hunk 4]
```diff
@@ -0,0 +1,46 @@
+-----[header]-----
+Kotlin Plan Generation - Particles
```
Comment (alxrsngrtn) [inline src/tools/tests/goldens/plan-generator-handles.cgtest:2]: s/Paritcles/Handles/?

File: src/tools/tests/goldens/plan-generator-handleConnections.cgtest [hunk 5]
```diff
@@ -0,0 +1,137 @@
+-----[header]-----
+Kotlin Plan Generation - Particles
```
Comment (alxrsngrtn) [inline src/tools/tests/goldens/plan-generator-handleConnections.cgtest:2]: s/Particles/Handle Connections/

Comment (alxrsngrtn) [reply src/tools/tests/goldens/plan-generator-handleConnections.cgtest:2]: ?

File: src/tools/tests/goldens/plan-generator-plans.cgtest [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+                    listOf(Annotation("persistent", emptyMap()))
+                )
+            )
+        )
+    ),
+    listOf(R_Handle0, R_Handle1),
+    emptyList()
+)
+
+-----[require]-----
```
Comment (alxrsngrtn) [inline src/tools/tests/goldens/plan-generator-plans.cgtest:328]: I like the require blocks; I haven't seen them before.

File: src/tools/tests/goldens/plan-generator-handleConnections.cgtest (hunk 2 above)
Comment (shans) [reply src/tools/tests/goldens/plan-generator-handleConnections.cgtest:120]: I don't think that fixes the issue - the point is that this is a singleton type variable, not an entity or a reference. I have a fix, I'll push it up now.

Comment (piotrswigon) [reply src/tools/tests/goldens/plan-generator-handleConnections.cgtest:120]: Ah, right. The PR I sent out fixes the wrapping of Entity/Ref types into a SingletonType when we generate the type for example for a handle, I think we should merge that as well.

**Pull request merged**